build/config/SignToolData.json

```diff
@@ -100,46 +103,82 @@
         "Vsix\\VisualStudioSetup.Next\\Roslyn.VisualStudio.Setup.Next.vsix",
         "Vsix\\VisualStudioSetup\\Roslyn.VisualStudio.Setup.vsix",
         "Vsix\\Roslyn\\RoslynDeployment.vsix",
-        "Vsix\\Templates\\Roslyn SDK.vsix"
+        "Vsix\\Templates\\Roslyn SDK.vsix",
+        "Vsix\\CodeAnalysisCompilers\\Microsoft.CodeAnalysis.Compilers.vsix"
+      ]
+    },
+    {
+      "certificate": null,
+      "strongName": null,
+      "values": [
+        "Nuget\\PerBuildPreRelease\\*.nupkg",
+        "Nuget\\PreRelease\\*.nupkg",
+        "Nuget\\Release\\*.nupkg"
       ]
     }
   ],
 
   "exclude": [
```

Comment:
Is there a better way to manage this list? It's getting pretty big and ugly...

Reply:
Which part: the set of nuget packages or the totality of the file?

Reply:
FWIW for the core repo's we drop a requires_signing file (https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/sign.targets#L86) in the bin folder for any binary that we need to sign. That file contains the information on which strong and and certificate to sign it with. Then as part of our signing step we just find all the requires_signing files and batch sign them. That prevents us form needing to maintain a list of all the files we need to sign.

Reply:
@weshaggard 

> That prevents us form needing to maintain a list of all the files we need to sign.

We prefer this approach: an explicit manifest of items that need to be signed.

Reply:
@jaredpar my concern is this seems to be yet another specification of our dependencies, i.e. any time @agocke takes a dependency on a new façade, we also have to add it to this exclusion list. Can we auto-exclude things already signed, or some other heuristic?

Reply:
Gotcha it's the exclusion list you're concerned about. 

> Can we auto-exclude things already signed, or some other heuristic?

The goal of the exclusion list is to help prevent signing errors by fully validating our assets at build time. It's very easy for a new binary to sneak into a VSIX that we didn't intend through innocuous changes to MSBuild files. If that binary is unsigned then suddenly the VSIX is not fully valid anymore. The exclusion list is meant to help with that by identifying the files which are deliberately meant to be there. 

I agree it's clunky though. It's also inexact because it uses file names which is not a strong, or unique, ID. I've considered a couple of solutions: 

1. Allow globs: `System.*`, `Microsoft.Build.*` and we're done ... right? Sounds good until you get into the details. Excluding `System.*` is easy sure but what about `Microsoft.Build.*`? That means we've excluded checking for our own packages. Hence we now need positive and negative excludes. At this point the list is hard to follow and easy for mistakes to happen. Not in favor of this option. 
1. Don't warn if the item is already signed: here we allow binaries which are fully signed to be added to the VSIX / NUPKG without any alteration of the configuration file. This means we still need to list items like NUPKG which can't be signed yet so the exclude list can't go away completely. Overall it's a bit more implicit but may be a good balance.

Reply:
Would there be some useful feature for "exclude everything that we're consuming from NuGet packages?" Ignoring the details of implementing such a check (you can probably figure it out via groveling either our project.assets.json or the packages directory or both), that means we're more pass-through.

I don't think we can entirely get rid of the exclusion list, but right now it seems having to update it is the norm, not the exception?

Reply:
NuGet content isn't sufficient though. What happens if a downstream repo, or a 3rd party, starts publishing unsigned binaries?

Reply:
Then validate it's signed before excluding it automatically. ;-) I would have presumed this exclusion list just excludes us from signing it, it also validates it's already signed at the same time?

Reply:
Part of the problem here is finding an API that actually tells me if a binary is signed or not: strong and cert. Lacking a good API for that I had to go to an imprecise exclusion list. Given a reliable API for determining signing I would just allow implicit excludes for such binaries.